src/sage/combinat/integer_vectors_mod_permgroup.py

```diff
@@ -763,18 +794,195 @@ def __iter__(self):
             [3, 1, 1, 2]
             [3, 0, 2, 2]
             [2, 2, 2, 1]
+
+        Check that :issue:`36681` is fixed::
+
+            sage: G = PermutationGroup([], domain=[])
+            sage: I = IntegerVectorsModPermutationGroup(G, sum=0)
+            sage: list(iter(I))
+            [[]]
+
+        Check that :issue:`36681` is fixed::
+
+            sage: G = PermutationGroup([], domain=[])
+            sage: I = IntegerVectorsModPermutationGroup(G, sum=3)
+            sage: list(iter(I))
+            []
+
         """
+        # Special cases when domain is empty.
+        if self.n == 0:
+            if self._sum is not None and self._sum > 0:
+                # No empty vector can have positive sum.
+                return iter(())
+            else:
+                # Sum is allowed to be zero.  It does not matter what
+                # the maxpart is, the empty vector is a solution.
+                return iter([self([])])
+
+        # General case, nonempty domain.
         if self._max_part < 0:
             return self.elements_of_depth_iterator(self._sum)
         else:
-            SF = RecursivelyEnumeratedSet_forest((self([0]*(self.n), check=False),),
-                              lambda x : [self(y, check=False) for y in canonical_children(self._sgs, x, self._max_part)],
-                              algorithm='breadth')
+            SF = RecursivelyEnumeratedSet_forest(
+                (self([0]*(self.n), check=False),),
+                lambda x: [self(y, check=False)
+                           for y in canonical_children(
+                                   self._sgs, x, self._max_part)],
+                algorithm='breadth')
             if self._sum is None:
                 return iter(SF)
             else:
                 return SF.elements_of_depth_iterator(self._sum)
 
+    def cardinality(self):
+        r"""
+        Return the number of integer vectors in the set.
+
+        The number is computed using the cycle index theorem, which is
+        faster than listing the vectors.
```

Comment:
I think it can be left in - perhaps moved into ` ..NOTE` . It does carry useful info (if you only care about the number, don't try to list vectors)

Reply:
I don't really mind, but I don't see the point either - it is obvious that you wouldn't implement a formula if enumeration is faster, isn't it?

Reply:
I don't understand why a user would *not* benefit from knowing what is fast and what is not. Certainly that is more relevant from a user's viewpoint than which particular algorithm is being used. So perhaps we remove the cycle index theorem then.

Reply:
I disagree, but let's leave it at that, then.  Let's not argue about seven words.

Reply:
I will remove the information as requested.

Reply:
For some reason the default `rank` and `unrank` still talk about speed.

```
   This is the default (brute force) implementation from the category
   "EnumeratedSets()" which can be used when the method "__iter__" is
   provided. Its complexity is O(r), where r is the rank of "obj".
```

Reply:
Let me then just add my personal reason why I find "uses the Redfield–Pólya theorem" valuable information and "faster than plain enumeration" not.

The first gives me some rough information on how the result is obtained.  This can be helpful if I have to understand the code (eg., whether I can adapt it to other, similar classes) and also what I can expect from it performance-wise.  For example, if the doc string says: "We essentially use linear search to find the element of given rank.", this tells me immediately that, if I know a custom unranking algorithm, I will be able to improve the performance.  Also, the code might use some ad-hoc tricks to make it slightly faster, which will almost certainly make it harder to decipher what the underlying algorithm really is.

I don't see any information I could gain from "faster than plain enumeration".  If it would be slower than plain enumeration, I wouldn't use a (nontrivial) formula, would I?

Reply:
> I will remove the information as requested.

No, please don't do this. It is absolutely normal for Sage documentation to explain what mathematical tools are behind the implementation. 
If a user wants to understand how a piece of code works, it's not a good idea to leave it totally unexplained, just like a blackbox. Sage has users from all walks of mathematical and CS life, I don't know what percentage of them heard about the cycle index theorem, perhaps 5% or less. It's not a topic that comes up in every maths-related undergraduate curriculum. 

I'd even like to ask for a reference to the cycle index theorem to be added to the docstring.

Reply:
> I don't see any information I could gain from "faster than plain enumeration". If it would be slower than plain enumeration, I wouldn't use a (nontrivial) formula, would I?

Just imagine you didn't know what the cycle index theorem is all about. The choice of the algorithm might have been like "we chose an algorithm we knew, and implemented, just because we could". Only after an explanation that it's a huge speedup it'd made sense.